src/mbgl/text/shaping.cpp

```diff
@@ -111,17 +115,25 @@ void justifyLine(std::vector<PositionedGlyph>& positionedGlyphs,
     }
 }
 
-float determineAverageLineWidth(const std::u16string& logicalInput,
+float determineAverageLineWidth(const TaggedString& logicalInput,
                                 const float spacing,
                                 float maxWidth,
-                                const Glyphs& glyphs) {
+                                const GlyphMap& glyphMap) {
     float totalWidth = 0;
     
-    for (char16_t chr : logicalInput) {
-        auto it = glyphs.find(chr);
-        if (it != glyphs.end() && it->second) {
-            totalWidth += (*it->second)->metrics.advance + spacing;
+    for (std::size_t i = 0; i < logicalInput.length(); i++) {
+        const SectionOptions& section = logicalInput.getSection(i);
```

Comment:
nit: `getSection` returns `SectionOptions`. I find the method names of `TaggedString` a bit hard to follow.

Reply:
Yeah, I see your point. Might make more sense to call it something like `getFormattingOptions`? At this point I'm following the JS naming conventions, but if we decide on better names here I can propagate them back to JS.